shenyu-admin/src/main/java/org/apache/shenyu/admin/service/SelectorService.java

```diff
@@ -87,47 +87,58 @@ default int createOrUpdate(SelectorDTO selectorDTO) {
      * @return rows int
      */
     int create(SelectorDTO selectorDTO);
-    
+
     /**
      * update selector.
      *
      * @param selectorDTO {@linkplain SelectorDTO}
      * @return rows int
      */
     int update(SelectorDTO selectorDTO);
-    
+
     /**
      * update selective selector.
      *
      * @param selectorDO {@linkplain SelectorDO}
      * @return rows int
      */
     int updateSelective(SelectorDO selectorDO);
-    
+
     /**
      * delete selectors.
      *
      * @param ids primary key.
      * @return rows int
      */
     int delete(List<String> ids);
-    
+
     /**
      * find selector by id.
      *
      * @param id primary key.
      * @return {@linkplain SelectorVO}
      */
     SelectorVO findById(String id);
-    
+
     /**
      * find selector by name.
      *
      * @param name the name
      * @return selector do
+     * @deprecated sice 2.6.0  Deprecated. By querying under this condition, multiple data are usually obtained.
+     *              Therefore, it is recommended to: {@linkplain SelectorService#findListByName(java.lang.String)}
      */
+    @Deprecated
     SelectorDO findByName(String name);
```

Comment:
The findByName() method should return multiple entries, and the original usage of accepting one entry is incorrect. So I have marked @ Deprecated (not recommended for others to use and minimize inappropriate use).
At present, only one place has been called (`org.apache.shenyu.admin.listener.zookeeper.HttpServiceDiscovery # updateSelectorHandler`), and HttpServiceDiscovery has long been marked  @ Deprecated by others. Although this class may not have been actually called, I still handle compatibility. #4928